wsd-frontend/src/components/wsd/Meme/client/MemeThreeDotMenu.tsx

```diff
@@ -0,0 +1,66 @@
+'use client'
+
+import * as Icons from 'lucide-react'
+
+import { Button } from '@/components/shadcn/button'
+import { Overlay, OverlayContent, OverlayTrigger } from '@/components/shadcn/overlay'
+
+import { APIType, Includes } from '@/api'
+import { downloadAndFormatImage } from '@/lib/fileUtils'
+
+import { toast } from 'sonner'
+
+export default function MemeThreeDotMenu({
+  post,
+}: {
+  post: Includes<Includes<APIType<'Post'>, 'user', APIType<'User'>>, 'tags', APIType<'PostTag'>[]>
+}) {
+  const actions = [
+    {
+      label: 'Download',
+      icon: Icons.Download,
+      action: async () => {
+        try {
+          await downloadAndFormatImage(post.image, `${post.title}.png`, 'image/png', 1)
+          toast('Image downloaded successfully')
+        } catch {
+          toast('Unknown error downloading image')
+        }
+      },
+    },
+  ]
+
+  return (
+    <Overlay breakpoint="md">
+      <OverlayTrigger>
+        <a
+          className="flex items-center gap-1 p-2 rounded-md transition-colors text-gray-500 hover:bg-secondary bg-transparent"
+          aria-label="More"
+        >
+          <Icons.Ellipsis size={20} />
+        </a>
+      </OverlayTrigger>
+      <OverlayContent
+        className="z-50 min-w-[50px] w-full md:w-fit bg-black p-1 shadow-md rounded-md"
+        popoverContentProps={{
+          align: 'start',
```

Comment:
I think end aligned looks better since the popover remains in the scroll area instead of outside - what do you think here?

Reply:
I think start align looks better 100% because the icons look aligned with the three dots, and its more common. But your point that it stays over the main body instead going over to the side. Lets go end it looks still good.